modules/ROOT/pages/execution-engine.adoc

```diff
@@ -29,6 +29,10 @@ For connectors created with the Mule SDK, the SDK determines the most
 appropriate processing type based on how the connector is implemented. For
 details on that mechanism, refer to the xref:1.1@mule-sdk::index.adoc[Mule SDK documentation].
 
+== Optimizations
+
+There are certain conditions on flow executions where the runtime may ignore the proposed Processing Type for event processor. This is due internal optimizations.
```

Comment:
Is it possible to share any example conditions or a specific scenario in which this optimization happens?

Reply:
I think that this is too internal, with just mentioning that this could occur is the right amount of information for a customer.